### PR TITLE
UISAUTCOMP-113 Re-position advanced search button when there is too little space in the search pane.

### PR DESCRIPTION
**Pull request by BogdanDenis**
## Description
Re-position advanced search button when there is too little space in the search pane.

## Screenshots

https://github.com/folio-org/stripes-authority-components/assets/19309423/68feef46-a83b-4722-acb6-60809730f83c

## Issues
[UISAUTCOMP-113](https://folio-org.atlassian.net/browse/UISAUTCOMP-113)

### CODE REVIEW

File: lib/AuthoritiesSearchForm/AuthoritiesSearchForm.js [hunk 1]
```diff
@@ -193,12 +191,13 @@ const AuthoritiesSearchForm = ({
                 {intl.formatMessage({ id: 'stripes-authority-components.label.search' })}
               </Button>
             </div>
-            <Row between="xs">
-              <Col xs={12} lg={6}>
+            <div className={css.resetAndAdvancedSearchWrapper}>
+              <div className={css.resetButtonWrapper}>
```
Comment (Dmytro-Melnyshyn) [inline lib/AuthoritiesSearchForm/AuthoritiesSearchForm.js:195]: Have you tried to add styles via Button's prop `buttonClass`?

File: lib/AuthoritiesSearchForm/AuthoritiesSearchForm.css [hunk 2]
```diff
@@ -7,3 +7,18 @@
 .searchField {
   margin-bottom: calc(var(--control-margin-bottom) / -2);
 }
+
+.resetAndAdvancedSearchWrapper {
+  display: flex;
+  flex-wrap: wrap;
+  gap: 0.5rem;
+  justify-content: space-between;
```
Comment (Dmytro-Melnyshyn) [inline lib/AuthoritiesSearchForm/AuthoritiesSearchForm.css:15]: Please check if it works correctly without `justify-content`.

File: lib/AuthoritiesSearchForm/AuthoritiesSearchForm.js (hunk 1 above)
Comment (BogdanDenis) [reply lib/AuthoritiesSearchForm/AuthoritiesSearchForm.js]: Updated styles

**Review by Dmytro-Melnyshyn — APPROVED** [review]

**Review by zburke — APPROVED** [review]
I'm approving because, yes, this does the job, BUT: it is feels deeply wrong to me that we need custom CSS to handle this kind of element-spacing glitch. @JohnC-80, do our buttons really not auto-resize to contain their text, and auto-reflow? 

140px is maybe enough for our medium-sized "Advanced search" button, but what if we change the locale to German and becomes a giant-sized "Überobersüperklikenwrapper fur Technik MARC grepin" button?

**Comment by BogdanDenis** [comment]
Hey @zburke that's a good point about how the button text length can change in other translations
I've tried it locally and it mostly is ok - resizing works when there is not enough space for the text, but there is an issue when there is not enough text in the button

https://github.com/folio-org/stripes-authority-components/assets/19309423/634c47de-4d1c-4d78-8b96-7bb0ec9f2129

We could set `text-wrap: wrap` but it looks kind of ugly
image

**Pull request merged**